gcloud/bigquery/job.py

```diff
@@ -615,7 +615,7 @@ def from_api_repr(cls, resource, client):
         dest_config = config['destinationTable']
         dataset = Dataset(dest_config['datasetId'], client)
         destination = Table(dest_config['tableId'], dataset)
-        source_urls = config['sourceUris']
+        source_urls = config.get('sourceUris', ())
```

Comment:
Yup, while testing `Table.upload_from_file`:  the job it creates in the background is a `load` job, but has no source URIs associated with it, since the data isn't ever in gCloud Storage.